src/Dev/Deprecation.php

```diff
@@ -228,24 +250,22 @@ public static function outputNotices(): void
         if (!self::isEnabled()) {
             return;
         }
-        $outputMessages = [];
-        // using a while loop with array_shift() to ensure that self::$userErrorMessageBuffer will have
-        // have values removed from it before calling user_error()
-        while (count(self::$userErrorMessageBuffer)) {
+
+        $count = 0;
+        $origCount = count(self::$userErrorMessageBuffer);
+        while ($origCount > $count) {
```

Comment:
As noted in a comment below, with non-numeric keys `array_shift()` was leaving an item in the array, so this just infinitely looped over re-outputting the final deprecation.